### PR TITLE
She 18 user login access control

### PR DESCRIPTION
**Pull request by hoonboon**
Current program status:
- Implemented user login and logout feature.
- Still pending for the remote secured API url for the secured content access feature.

### CODE REVIEW

File: www/js/constants.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	 * Handling CORS issues for 'ionic serve'
+	 * 
+	 * Ref: http://blog.ionic.io/handling-cors-issues-in-ionic/
+	 * Proxied path is defined in ionic.config.json
+	 * 
+	 * Options:
+	 * - true: when need to test run using 'ionic serve' or 'ionic run -l'
+	 * - false: when test run using 'ionic run' or 'ionic emulate' or during UAT or Production
+	 */ 
+	useProxy : false,
```
Comment (hutingung) [inline www/js/constants.js:14]: Usually we use "CORS" plugin at the browser to test the issue. But it is good also to use this out of the box function from ionic. Is it documented into the README?

File: www/js/controllers/epaper-menuctrl.js [hunk 2]
```diff
@@ -58,6 +58,203 @@ angular.module('epaper.controllers')
...
+				
+			} else if (sessionStorage.getItem('toRefreshShApiSession') == undefined) {
+				
+				var refreshShApiSession = function() {
+					
+					ePaperService.login(
+							localStorage.getItem('shApiUsername'), 
+							window.atob(localStorage.getItem('shApiPassword'))).then(function(response){
+						
+						console.log('refreshShApiSession(): data=' + JSON.stringify(response.data));
```
Comment (hutingung) [inline www/js/controllers/epaper-menuctrl.js]: remove all the console.

File: www/js/controllers/epaper-menuctrl.js [hunk 3]
```diff
@@ -58,6 +58,203 @@ angular.module('epaper.controllers')
...
+			if (ShApiConstants.useProxy) {
+				$cookies.remove(localStorage.getItem('shApiSessionKey'), {path:'/'});
+			}
+			
+		} else {
+			
+			// verify if session cookies is still available
+			// - if gone need to refresh session data based on localStorage values
+			// - possible scenario: user closed the app intentionally or restart device
+			if (ShApiConstants.useProxy) {
```
Comment (hutingung) [inline www/js/controllers/epaper-menuctrl.js:83]: Can you explain more on the useProxy? It is written in the Constants.js that useProxy is to handle CORS. why we use it to clear the cookies storage?

Comment (hoonboon) [reply www/js/controllers/epaper-menuctrl.js:83]: Referring to my comments in Jira for SHE18, cookies need be explicitly managed only when running on 'ionic serve'. 

The constant 'useproxy' determines if currently running with 'ionic serve' or not.

File: www/js/constants.js (hunk 1 above)
Comment (hoonboon) [reply www/js/constants.js:14]: Will add some notes to the readme.

File: www/js/controllers/epaper-menuctrl.js (hunk 2 above)
Comment (hoonboon) [reply www/js/controllers/epaper-menuctrl.js]: ~~Added this into SHE-13 Code Refactor~~
Committed changes to this same branch.

File: www/js/constants.js (hunk 1 above)
Comment (hoonboon) [reply www/js/constants.js:14]: Notes added in the files `README.md` and `developer-guide-macosx.md`.

File: www/js/controllers/epaper-menuctrl.js (hunk 3 above)
Comment (hutingung) [reply www/js/controllers/epaper-menuctrl.js:83]: I am ok with it currently even I am not comfortable with it. Ideally development stuff (ionic serve) should not polluted the actual application.

**Comment by hutingung** [comment]
Please resolve the conflict first.

**Pull request merged**